### PR TITLE
Adding financial metrics function

### PR DESCRIPTION
**Pull request by dorisyycai**
Adding a function to calculate the number of returns, average, total return, standard deviation and sharp ratio of the given return series.

Please review and let me know if you see any issue.

### CODE REVIEW

File: src/quanteda/financial_metrics.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+def generate_financial_metrics(random_returns_df, annual_risk_free=0.00):
+    """
+    Calculate financial metrics based on a DataFrame of random returns on time series.
+
+    Parameters
+    ----------
+    random_returns_df : pd.DataFrame
+        DataFrame containing random return values with a datetime index.
+
+    annual_risk_free : float, default is 0.00.
```
Comment (JohnShiuMK) [inline src/quanteda/financial_metrics.py:12]: to match the Pandas format

```suggestion
    annual_risk_free : float, default 0.00
```

File: src/quanteda/financial_metrics.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        Annualized risk-free rate as a decimal (e.g., 0.02 for 2%).
+        
+    Returns
+    -------
+    pd.DataFrame:
```
Comment (JohnShiuMK) [inline src/quanteda/financial_metrics.py:17]:
```suggestion
    pd.DataFrame
```

File: src/quanteda/financial_metrics.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    Examples
+    --------
+    >>> random_returns_df = generate_return_series(expected_annual_return, 
+                                                   annual_volatility, 
+                                                   freq='H', 
+                                                   length=5, 
+                                                   num_series=3, 
+                                                   dist="normal")
+    >>> annual_risk_free_rate = 0.02
+    >>> result = generate_financial_metrics(random_returns_df, annual_risk_free_rate)
```
Comment (JohnShiuMK) [inline src/quanteda/financial_metrics.py:30]: simplify the example 

```suggestion
    >>> generate_financial_metrics(random_returns_df, annual_risk_free_rate=0.02)
```

File: src/quanteda/financial_metrics.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    >>> print(result)
```
Comment (JohnShiuMK) [inline src/quanteda/financial_metrics.py:31]:
```suggestion
```

File: src/quanteda/financial_metrics.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+    Examples
+    --------
+    >>> random_returns_df = generate_return_series(expected_annual_return, 
+                                                   annual_volatility, 
+                                                   freq='H', 
+                                                   length=5, 
+                                                   num_series=3, 
+                                                   dist="normal")
+    >>> annual_risk_free_rate = 0.02
```
Comment (JohnShiuMK) [inline src/quanteda/financial_metrics.py:29]:
```suggestion
```

**Review by JohnShiuMK — CHANGES_REQUESTED** [review]
the others are good!

**Comment by dorisyycai** [comment]
Thanks @JohnShiuMK. Agreed with all the suggested changes. Also adding @meretelutz @jbarns14 for review. Once approved, I can merge to the main branch. Thank you.

**Review by meretelutz — APPROVED** [review]
Great job! Approving this for merging.

**Review by JohnShiuMK — COMMENTED** [review]
Good

**Review by JohnShiuMK — APPROVED** [review]

**Comment by JohnShiuMK** [comment]
Thanks all

**Pull request merged**